verticapy/tests_new/performance/vertica/test_qprof.py

```diff
@@ -0,0 +1,1384 @@
+        ],
+    )
+    @pytest.mark.parametrize(
+        "metric",
+        [
+            "cost",
+            "rows",
+            "exec_time_ms",
+            "est_rows",
+            "proc_rows",
+            "prod_rows",
+            "rle_prod_rows",
+            "clock_time_us",
+            # "cstall_us",  # ZeroDivisionError
+            # "pstall_us",  # ZeroDivisionError
+            "mem_res_mb",
+            # "mem_all_mb",  # ZeroDivisionError
+            # ["cost", "proc_rows"],
+        ],
+    )
+    def test_get_qplan_tree(
+        self,
+        path_id,
+        path_id_info,
+        show_ancestors,
+        metric,
+        pic_path,
+        return_graphviz,
+    ):
+        """
+        test function for get_qplan_tree
+        """
+        # need to check. get_qplan_tree is not returning anything with transaction_id, statement_id
+        qprof = QueryProfiler(transactions=QPROF_SQL2)
+        qprof_qplan_tree = qprof.get_qplan_tree(
+            path_id=path_id,
+            path_id_info=path_id_info,
+            show_ancestors=show_ancestors,
+            metric=metric,
+            pic_path=pic_path,
+            return_graphviz=return_graphviz,
+        )
+        # actual
+        ss1 = ""
+        if return_graphviz:
+            graph = Digraph()
+            source_lines = str(qprof_qplan_tree).splitlines()
+            # Remove 'digraph tree {'
+            source_lines.pop(0)
+            # Remove the closing brackets '}'
+            source_lines.pop(-1)
+            # Append the nodes to body
+            graph.body += source_lines
+        else:
+            graph = qprof_qplan_tree
+
+        json_string = graph.pipe("json").decode()
+        json_dict = json.loads(json_string)
+        for obj in json_dict["objects"]:
+            if "tooltip" in obj.keys():
+                _ss1 = ""
+                for s in obj["tooltip"].split("\n"):
+                    if s != "" and not s.startswith(
+                        (
+                            "cost",
+                            "rows",
+                            "exec_time_ms",
+                            "est_rows",
+                            "proc_rows",
+                            "prod_rows",
+                            "rle_prod_rows",
+                            "clock_time_us",
+                            "cstall_us",
+                            "pstall_us",
+                            "mem_res_mb",
+                            "mem_all_mb",
+                        )
+                    ):
+                        _ss1 += s + ""
+                    # for return_graphviz = True
+                    if return_graphviz and metric in s:
+                        _ss1 = re.search(rf".+{metric}", s).group(0).replace(metric, "")
+                ss1 += _ss1.strip().replace('"', "'") + " "
+        print("<<<<<<<<<<<<<<<<<<< Actual result >>>>>>>>>>>>>>>>>>>>>>")
+        print(ss1)
+
+        # expected
+        ss2 = ""
```

Comment:
What is the expected and observed in this function (`test_get_qplan_tree`?) 

I see lots of string manipulation that might help to line things up, but I'm not really sure what the inputs and outputs are. It would help to have a comment describing what the expected and observed are, and why some cleanup is required.

Reply:
I will add comments to each test to make it more readable.